runtime/resolvers/metricsview_cache_key.go

```diff
@@ -0,0 +1,159 @@
+	"github.com/rilldata/rill/runtime/pkg/mapstructureutil"
+)
+
+var errCachingDisabled = errors.New("metrics_cache_key: caching is disabled")
+
+func init() {
+	runtime.RegisterResolverInitializer("metrics_cache_key", newMetricsViewCacheKeyResolver)
+}
+
+type metricsViewCacheKeyResolver struct {
+	runtime    *runtime.Runtime
+	instanceID string
+	mvName     string
+	mv         *runtimev1.MetricsViewSpec
+	streaming  bool
+	exectuor   *metricsview.Executor
+	args       *metricsViewCacheKeyResolverArgs
+}
+
+type metricsViewCacheKeyResolverArgs struct {
+	Priority int `mapstructure:"priority"`
+}
+
+type metricsViewCacheKey struct {
+	MetricsView string `mapstructure:"metrics_view"`
+}
+
+func newMetricsViewCacheKeyResolver(ctx context.Context, opts *runtime.ResolverOptions) (runtime.Resolver, error) {
+	tr := &metricsViewCacheKey{}
+	if err := mapstructureutil.WeakDecode(opts.Properties, tr); err != nil {
+		return nil, err
+	}
+
+	args := &metricsViewCacheKeyResolverArgs{}
+	if err := mapstructureutil.WeakDecode(opts.Args, args); err != nil {
+		return nil, err
+	}
+
+	ctrl, err := opts.Runtime.Controller(ctx, opts.InstanceID)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := ctrl.Get(ctx, &runtimev1.ResourceName{Kind: runtime.ResourceKindMetricsView, Name: tr.MetricsView}, false)
+	if err != nil {
+		return nil, err
+	}
+
+	mv := res.GetMetricsView().State.ValidSpec
+	if mv == nil {
+		return nil, fmt.Errorf("metrics view %q is invalid", res.Meta.Name.Name)
+	}
+
+	security, err := opts.Runtime.ResolveSecurity(opts.InstanceID, opts.Claims, res)
+	if err != nil {
+		return nil, err
+	}
+
+	if !security.CanAccess() {
+		return nil, runtime.ErrForbidden
+	}
+
+	executor, err := metricsview.NewExecutor(ctx, opts.Runtime, opts.InstanceID, mv, res.GetMetricsView().State.Streaming, security, args.Priority)
+	if err != nil {
+		return nil, err
+	}
+
+	return &metricsViewCacheKeyResolver{
+		runtime:    opts.Runtime,
+		instanceID: opts.InstanceID,
+		mvName:     tr.MetricsView,
+		streaming:  res.GetMetricsView().State.Streaming,
+		exectuor:   executor,
+		mv:         mv,
+		args:       args,
+	}, nil
+}
+
+func (r *metricsViewCacheKeyResolver) Close() error {
+	return nil
+}
+
+func (r *metricsViewCacheKeyResolver) Cacheable() bool {
+	return true
+}
+
+func (r *metricsViewCacheKeyResolver) CacheKey(ctx context.Context) ([]byte, bool, error) {
+	var sb strings.Builder
+	sb.WriteString(runtime.ResourceKindMetricsView)
+	sb.WriteString(":")
+	sb.WriteString(r.mvName)
+	sb.WriteString(":")
+	sb.WriteString("cahe_key")
+	ttlSeconds := r.mv.CacheKeyTtlSeconds
+	if ttlSeconds == 0 && r.streaming {
+		// If streaming, we need to cache the key for 60 seconds
+		// For non streaming metrics view we don't need to expire the key as data itself will be invalidated basis the ref's state version
+		ttlSeconds = 60
+	}
+	if ttlSeconds != 0 {
+		sb.WriteString(":")
+		sb.WriteString(truncateTime(time.Now(), r.mv.CacheKeyTtlSeconds).Format(time.RFC3339))
```

Comment:
It's not using `ttlSeconds`:

```suggestion
		sb.WriteString(truncateTime(time.Now(), ttlSeconds).Format(time.RFC3339))
```